### PR TITLE
avm1: Correct `WaitForFrame` and `WaitForFrame2` actions

### PR DESCRIPTION
**Pull request by Toad06**
This PR fixes the following cases:

`WaitForFrame` action:
- `ifFrameLoaded(n)` (where n is an integer equals to `_framesloaded`) evaluates to true.
- Any frame number above 16000 causes this action to evaluate to false.

`WaitForFrame2` action:
- `ifFrameLoaded(_framesloaded)` and `ifFrameLoaded(_framesloaded + 1)` evaluate to true.
- `ifFrameLoaded(2147483647 + 1)` evaluates to false.

---

I added two separate tests for these actions. A test already existed for `WaitForFrame2` but I replaced it because it didn't seem to test the correct values sometimes, for instance:

```
ifFrameLoaded(0) { trace("/// frame -43 marked as loaded"); }
ifFrameLoaded(NaN + 0) { trace("/// frame NaN marked as loaded"); }
```

### CODE REVIEW

File: core/src/avm1/activation.rs [hunk 1]
```diff
@@ -2305,16 +2305,23 @@ impl<'a, 'gc> Activation<'a, 'gc> {
         action: WaitForFrame,
         r: &mut Reader<'_>,
     ) -> Result<FrameControl<'gc>, Error<'gc>> {
-        let loaded = self
-            .target_clip()
-            .and_then(|dobj| dobj.as_movie_clip())
-            .map(|mc| mc.frames_loaded() > min(action.frame, mc.total_frames() - 1))
-            .unwrap_or(true);
+        let frame_num = action.frame;
+        let loaded = if frame_num > 16000 {
```
Comment (Toad06) [inline core/src/avm1/activation.rs:2309]: I guess it would make sense to define this value as a constant. But not sure what could be best place for this.

**Comment by torokati44** [comment]
I can't test right now, but does this perhaps fix some more of https://github.com/ruffle-rs/ruffle/issues/6144?

**Comment by Toad06** [comment]
Unfortunately it doesn't, Ruffle still loads more frames than Flash Player on startup, which causes `ifFrameLoaded(n)` to evaluate differently.

**Review by Dinnerbone — APPROVED** [review]

**Review by adrian17 — APPROVED** [review]
Could that mean that in reality, `_framesloaded` is the one that lies by 1? As in, it sounds more likely that `_framesloaded` reports a lower number of loaded frames than the actual number, than that `ifFrameLoaded` lies about a frame being loaded that's not actually loaded.
(at least that's how I understand it)

**Comment by Toad06** [comment]
> Could that mean that in reality, `_framesloaded` is the one that lies by 1? As in, it sounds more likely that `_framesloaded` reports a lower number of loaded frames than the actual number, than that `ifFrameLoaded` lies about a frame being loaded that's not actually loaded. (at least that's how I understand it)

This looks accurate. I added traces to z0r 916 ([edited file here](https://github.com/ruffle-rs/ruffle/files/11750517/z916.zip)) and this is the output I got from the Flash Player projector:

Frame 1:
> _framesloaded: 0
>_currentframe: 1

Frame 42:
> _framesloaded: 42
> _currentframe: 42

**Pull request merged**